app/models/calculator/weight.rb

```diff
@@ -54,6 +54,8 @@ def weight_per_final_weight_volume(line_item)
     #    Customer ends up getting 350mL (line_item.final_weight_volume) of wine
     #      that represent 2.8 (quantity_implied_in_final_weight_volume) glasses of wine
     def quantity_implied_in_final_weight_volume(line_item)
+      return line_item.quantity if line_item.variant.unit_value.zero?
```

Comment:
I looked at au-prod:

```rb
Spree::Variant.where(unit_value: 0).count
# => 38

Spree::Variant.where(unit_value: nil).count
# => 630
```

I think we can catch these cases as well:

```suggestion
      return line_item.quantity if line_item.variant.unit_value.to_i.zero?
```

What do you think? Is that case covered somewhere else?

Reply:
I'm not sure if this is a valid case but we do have data like this:
```rb
item = Spree::LineItem.where("final_weight_volume > 0").joins(:variant).where("spree_variants.unit_value is null").last
Calculator::Weight.new.compute(item)
# TypeError: nil can't be coerced into BigDecimal
```